contrib/yaml/postgresql-10.yaml

```diff
@@ -1,289 +1,252 @@
+          description: pg_stat_database_temp_files
+        - name: temp_bytes
+          type: gauge
+          description: pg_stat_database_temp_bytes
+        - name: tup_returned
+          type: counter
+          description: pg_stat_database_tup_returned
+        - name: tup_fetched
+          type: counter
+          description: pg_stat_database_tup_fetched
+        - name: tup_inserted
+          type: counter
+          description: pg_stat_database_tup_inserted
+        - name: tup_updated
+          type: counter
+          description: pg_stat_database_tup_updated
+        - name: tup_deleted
+          type: counter
+          description: pg_stat_database_tup_deleted
+        - name: xact_commit
+          type: counter
+          description: pg_stat_database_xact_commit
+        - name: xact_rollback
+          type: counter
+          description: pg_stat_database_xact_rollback
+        - name: conflicts
+          type: counter
+          description: pg_stat_database_conflicts
+        - name: numbackends
+          type: gauge
+          description: pg_stat_database_numbackends
     tag: pg_stat_database
     sort: data
-    columns:
-      -
-        name: database
-        type: label
-      -
-        name: blk_read_time
-        type: gauge
-        description: pg_stat_database_blk_read_time
-      -
-        name: blk_write_time
-        type: gauge
-        description: pg_stat_database_blk_write_time
-      -
-        name: blks_hit
-        type: gauge
-        description: pg_stat_database_blks_hit
-      -
-        name: blks_read
-        type: gauge
-        description: pg_stat_database_blks_read
-      -
-        name: deadlocks
-        type: gauge
-        description: pg_stat_database_deadlocks
-      -
-        name: temp_files
-        type: gauge
-        description: pg_stat_database_temp_files
-      -
-        name: temp_bytes
-        type: gauge
-        description: pg_stat_database_temp_bytes
-      -
-        name: tup_returned
-        type: gauge
-        description: pg_stat_database_tup_returned
-      -
-        name: tup_fetched
-        type: gauge
-        description: pg_stat_database_tup_fetched
-      -
-        name: tup_inserted
-        type: gauge
-        description: pg_stat_database_tup_inserted
-      -
-        name: tup_updated
-        type: gauge
-        description: pg_stat_database_tup_updated
-      -
-        name: tup_deleted
-        type: gauge
-        description: pg_stat_database_tup_deleted
-      -
-        name: xact_commit
-        type: gauge
-        description: pg_stat_database_xact_commit
-      -
-        name: xact_rollback
-        type: gauge
-        description: pg_stat_database_xact_rollback
-      -
-        name: conflicts
-        type: gauge
-        description: pg_stat_database_conflicts
-      -
-        name: numbackends
-        type: gauge
-        description: pg_stat_database_numbackends
+    collector: stat_db
```

Comment:
We can probably prefix the collector names of internal queries with something like `pgexporter_*` and raise an error if they use the prefix `pgexporter_` in their collector names. Will that work?

Reply:
Nah, I don't like that -- but I agree, there are reserved collector words

Reply:
Hmmm, should I consider a prefix? A prefix would also enable us to be more flexible, and have the possibility of adding collectors in future without risking conflicting with user defined collectors. Can be something like a dunder (`__`) as well (as prefix and suffix)

I can't think of any other than releasing a list of reserved collector words (like language reserved words)

Reply:
Why is this important right now ? We can circle back to it, and make sure that it works for 0.4.

Only the internal names are important to document if people aren't supplying a YAML file

Reply:
Hmm understood. I was suggesting special collector names so as to exempt them from the `collector_pass` check function I had added that checks if a collector is enabled or not.

That way some queries will always be run, and some dependent on the user. I'll explore a bit more on this topic.

Reply:
I think, if we want some queries to always run, and assuming they're all internal queries, I can expand the yaml-from-file-pointer functionality to take in an argument that is true if the query is always allowed.

So then, for those queries I can disable the collector-pass check.

How do you feel about this approach to the problem? If you want I can give a following commit with the changes I am talking about

Reply:
The "General" metrics should always be included - https://pgexporter.github.io/metrics.html - and they could be different between versions.

Reply:
Yes, you'd mentioned this while I was working on the collector feature, so [general queries are always run](https://github.com/pgexporter/pgexporter/blob/main/src/libpgexporter/prometheus.c#L568) as there is no collector pass in them

Reply:
I feel like this a separate dicussion - a user should just be able to specify a list of collectors, and a version - then the rest is automatic

Reply:
Also other metrics that are included in general as well like [server information](https://github.com/pgexporter/pgexporter/blob/main/src/libpgexporter/prometheus.c#L589), [version information](https://github.com/pgexporter/pgexporter/blob/main/src/libpgexporter/prometheus.c#L630), [primary information]( https://github.com/pgexporter/pgexporter/blob/main/src/libpgexporter/prometheus.c#L766) and [core information](https://github.com/pgexporter/pgexporter/blob/main/src/libpgexporter/prometheus.c#L841).

Other than that all of the queries have a collector-pass in them like [this](https://github.com/pgexporter/pgexporter/blob/main/src/libpgexporter/prometheus.c#L1128)

Reply:
> I feel like this a separate dicussion - a user should just be able to specify a list of collectors, and a version - then the rest is automatic

Got it. I'll keep that in mind. I feel this would go weel when I'm creating separate PRs for each version, where I'll address this issue as well as add any other metrics etc. for each version in each. Or do you want me to address this issue here?

Reply:
Separate is best

Reply:
So do I need to make any further changes to this PR?

Reply:
I'll do another pass tomorrow

Reply:
Ok! Looking forward to finishing this and moving on to the suggestions you mentioned. I'm quite curious about them